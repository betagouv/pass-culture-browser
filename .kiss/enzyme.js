import React from 'react'
import { shallow } from 'enzyme'

import MyComponent from '../MyComponent'

describe('src | components | MyComponent', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      const props = { errors: {} }
      const wrapper = shallow(<MyComponent {...props} />)
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
