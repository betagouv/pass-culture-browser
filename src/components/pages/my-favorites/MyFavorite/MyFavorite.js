import PropTypes from 'prop-types'
import React from 'react'
import { Link } from 'react-router-dom'

const MyFavorite = ({ name, offerVersoUrl, thumbUrl }) => {
  return (
    <li className="mf-my-favorite">
      <Link className="mf-link"
            to={offerVersoUrl}>
        <div className="mf-thumb">
          {
            thumbUrl && <img alt="" src={thumbUrl}/>
          }
        </div>
        <div className="mf-infos">
          <div className="mf-heading">
            <div className="mf-name">{name}</div>
          </div>
        </div>
      </Link>
    </li>
  )
}

MyFavorite.defaultProps = {
  stringifyDate: 'Permanent',
  thumbUrl: null,
}

MyFavorite.propTypes = {
  name: PropTypes.string.isRequired,
  offerVersoUrl: PropTypes.string.isRequired,
  thumbUrl: PropTypes.string,
}

export default MyFavorite
