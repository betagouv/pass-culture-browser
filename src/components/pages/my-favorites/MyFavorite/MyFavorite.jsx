import PropTypes from 'prop-types'
import React from 'react'
import { Link } from 'react-router-dom'

import Icon from '../../../layout/Icon'

const MyFavorite = ({
  humanizeRelativeDistance,
  name,
  offerTypeLabel,
  status,
  versoUrl,
  thumbUrl,
}) => (
  <li className="mf-my-favorite">
    <Link
      className="teaser-link"
      to={versoUrl}
    >
      <div className="teaser-thumb">{thumbUrl && <img
        alt=""
        src={thumbUrl}
                                                 />}
      </div>
      <div className="teaser-wrapper mf-wrapper">
        <div className="teaser-title">{name}</div>
        <div className="teaser-sub-title">{offerTypeLabel}</div>
        <div className="mf-infos">
          {status && <span className={`mf-status mf-${status.class}`}>{status.label}</span>}
          <span className="teaser-sub-title">{humanizeRelativeDistance}</span>
        </div>
      </div>
      <div className="teaser-arrow">
        <Icon
          className="teaser-arrow-img"
          svg="ico-next-S"
        />
      </div>
    </Link>
  </li>
)

MyFavorite.defaultProps = {
  status: null,
  thumbUrl: null,
}

MyFavorite.propTypes = {
  humanizeRelativeDistance: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  offerTypeLabel: PropTypes.string.isRequired,
  status: PropTypes.shape(),
  thumbUrl: PropTypes.string,
  versoUrl: PropTypes.string.isRequired,
}

export default MyFavorite
