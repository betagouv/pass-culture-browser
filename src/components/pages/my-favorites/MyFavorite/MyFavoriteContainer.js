import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'
import { compose } from 'redux'

import MyFavorite from './MyFavorite'
import { getHumanizeRelativeDistance } from '../../../../utils/geolocation'
import selectFirstMatchingBookingByStocks from '../../../../selectors/selectFirstMatchingBookingByStocks'
import selectOfferById from '../../../../selectors/selectOfferById'
import getHumanizeRelativeDate from '../../../../utils/date/getHumanizeRelativeDate'

export const reservationStatus = (
  isFinished,
  isFullyBooked,
  hasBookings,
  isBooked,
  humanizeRelativeDate
) => {
  if (isFinished) {
    return {
      label: 'Terminé',
      class: 'finished',
    }
  } else if (isFullyBooked) {
    return {
      label: 'Épuisé',
      class: 'fully-booked',
    }
  } else if (hasBookings) {
    if (isBooked) {
      return {
        label: 'Réservé',
        class: 'booked',
      }
    } else {
      return {
        label: 'Annulé',
        class: 'cancelled',
      }
    }
  } else if (humanizeRelativeDate) {
    if (humanizeRelativeDate === 'Demain') {
      return {
        label: humanizeRelativeDate,
        class: 'tomorrow',
      }
    } else {
      return {
        label: humanizeRelativeDate,
        class: 'today',
      }
    }
  }

  return null
}

export const mapStateToProps = (state, ownProps) => {
  const { favorite } = ownProps
  const { offerId } = favorite
  const offer = selectOfferById(state, offerId)
  const { venue } = offer
  const firstMatchingBooking = selectFirstMatchingBookingByStocks(state)
  const isBooked = firstMatchingBooking ? !firstMatchingBooking.isCancelled : false
  const hasBookings = typeof firstMatchingBooking !== 'undefined'
  const { dateRange = [], isFinished, isFullyBooked } = offer || {}
  const offerBeginningDate = dateRange[0] || null
  const humanizeRelativeDate = offerBeginningDate
    ? getHumanizeRelativeDate(offerBeginningDate)
    : null
  const status = reservationStatus(
    isFinished,
    isFullyBooked,
    hasBookings,
    isBooked,
    humanizeRelativeDate
  )
  const humanizeRelativeDistance = getHumanizeRelativeDistance(
    state.geolocation.latitude,
    state.geolocation.longitude,
    venue.latitude,
    venue.longitude
  )
  const { pathname, search } = location
  const mediationId = favorite.mediationId ? `/${favorite.mediationId}` : ''
  const detailsUrl = `${pathname}/details/${offer.id}${mediationId}${search}`

  return {
    detailsUrl,
    humanizeRelativeDistance,
    offer,
    status,
  }
}

export default compose(
  withRouter,
  connect(mapStateToProps)
)(MyFavorite)
