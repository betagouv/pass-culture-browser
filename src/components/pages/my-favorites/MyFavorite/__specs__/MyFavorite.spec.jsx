import { shallow } from 'enzyme'
import React from 'react'
import { Link } from 'react-router-dom'

import MyFavorite from '../MyFavorite'

describe('src | components | pages | my-favorites | MyFavorite | MyFavorite', () => {
  let props

  beforeEach(() => {
    props = {
      detailsUrl: 'fake/url',
      favorite: {
        id: 'AE',
        thumbUrl: '',
      },
      humanizeRelativeDistance: '10 km',
      location: {
        pathname: '',
        search: '',
      },
      offer: {
        id: 'Fake id',
        name: 'Fake offer name',
        product: {
          offerType: {
            appLabel: 'Fake offer type label',
          },
        },
      },
      status: {
        label: 'Réservé',
        class: 'booked',
      },
    }
  })

  it('should match the snapshot', () => {
    // when
    const wrapper = shallow(<MyFavorite {...props} />)

    // then
    expect(wrapper).toMatchSnapshot()
  })

  it('should render a booking by default', () => {
    // when
    const wrapper = shallow(<MyFavorite {...props} />)

    // then
    const link = wrapper.find(Link)
    const img = wrapper.find('img')
    const title = wrapper.find('.teaser-title').text()
    const type = wrapper.find('.mf-wrapper > .teaser-sub-title').text()
    const booked = wrapper.find('.mf-booked').text()
    const distance = wrapper.find('.mf-infos .teaser-sub-title').text()
    expect(link).toHaveLength(1)
    expect(img).toHaveLength(0)
    expect(title).toBe('Fake offer name')
    expect(type).toBe('Fake offer type label')
    expect(booked).toBe('Réservé')
    expect(distance).toBe('10 km')
  })
})
