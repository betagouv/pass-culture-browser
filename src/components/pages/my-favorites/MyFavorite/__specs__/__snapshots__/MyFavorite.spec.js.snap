// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | my-favorites | MyFavorite | MyFavorite should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MyFavorite
    humanizeRelativeDistance="10 km"
    name="Fake offer name"
    offerTypeLabel="Fake offer type label"
    status={
      Object {
        "class": "booked",
        "label": "Réservé",
      }
    }
    thumbUrl={null}
    versoUrl="https://example.net/decouverte/ME/FA/verso"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        className="teaser-link"
        replace={false}
        to="https://example.net/decouverte/ME/FA/verso"
      >
        <div
          className="teaser-thumb"
        />
        <div
          className="teaser-wrapper mf-wrapper"
        >
          <div
            className="teaser-title"
          >
            Fake offer name
          </div>
          <div
            className="teaser-sub-title"
          >
            Fake offer type label
          </div>
          <div
            className="mf-infos"
          >
            <span
              className="mf-status mf-booked"
            >
              Réservé
            </span>
            <span
              className="teaser-sub-title"
            >
              10 km
            </span>
          </div>
        </div>
        <div
          className="teaser-arrow"
        >
          <Icon
            alt={null}
            className="teaser-arrow-img"
            src={null}
            svg="ico-next-S"
          />
        </div>
      </Link>,
      "className": "mf-my-favorite",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div
            className="teaser-thumb"
          />,
          <div
            className="teaser-wrapper mf-wrapper"
          >
            <div
              className="teaser-title"
            >
              Fake offer name
            </div>
            <div
              className="teaser-sub-title"
            >
              Fake offer type label
            </div>
            <div
              className="mf-infos"
            >
              <span
                className="mf-status mf-booked"
              >
                Réservé
              </span>
              <span
                className="teaser-sub-title"
              >
                10 km
              </span>
            </div>
          </div>,
          <div
            className="teaser-arrow"
          >
            <Icon
              alt={null}
              className="teaser-arrow-img"
              src={null}
              svg="ico-next-S"
            />
          </div>,
        ],
        "className": "teaser-link",
        "replace": false,
        "to": "https://example.net/decouverte/ME/FA/verso",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": null,
            "className": "teaser-thumb",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="teaser-title"
              >
                Fake offer name
              </div>,
              <div
                className="teaser-sub-title"
              >
                Fake offer type label
              </div>,
              <div
                className="mf-infos"
              >
                <span
                  className="mf-status mf-booked"
                >
                  Réservé
                </span>
                <span
                  className="teaser-sub-title"
                >
                  10 km
                </span>
              </div>,
            ],
            "className": "teaser-wrapper mf-wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Fake offer name",
                "className": "teaser-title",
              },
              "ref": null,
              "rendered": "Fake offer name",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Fake offer type label",
                "className": "teaser-sub-title",
              },
              "ref": null,
              "rendered": "Fake offer type label",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="mf-status mf-booked"
                  >
                    Réservé
                  </span>,
                  <span
                    className="teaser-sub-title"
                  >
                    10 km
                  </span>,
                ],
                "className": "mf-infos",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Réservé",
                    "className": "mf-status mf-booked",
                  },
                  "ref": null,
                  "rendered": "Réservé",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "10 km",
                    "className": "teaser-sub-title",
                  },
                  "ref": null,
                  "rendered": "10 km",
                  "type": "span",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Icon
              alt={null}
              className="teaser-arrow-img"
              src={null}
              svg="ico-next-S"
            />,
            "className": "teaser-arrow",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "alt": null,
              "className": "teaser-arrow-img",
              "src": null,
              "svg": "ico-next-S",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": [Function],
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          className="teaser-link"
          replace={false}
          to="https://example.net/decouverte/ME/FA/verso"
        >
          <div
            className="teaser-thumb"
          />
          <div
            className="teaser-wrapper mf-wrapper"
          >
            <div
              className="teaser-title"
            >
              Fake offer name
            </div>
            <div
              className="teaser-sub-title"
            >
              Fake offer type label
            </div>
            <div
              className="mf-infos"
            >
              <span
                className="mf-status mf-booked"
              >
                Réservé
              </span>
              <span
                className="teaser-sub-title"
              >
                10 km
              </span>
            </div>
          </div>
          <div
            className="teaser-arrow"
          >
            <Icon
              alt={null}
              className="teaser-arrow-img"
              src={null}
              svg="ico-next-S"
            />
          </div>
        </Link>,
        "className": "mf-my-favorite",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div
              className="teaser-thumb"
            />,
            <div
              className="teaser-wrapper mf-wrapper"
            >
              <div
                className="teaser-title"
              >
                Fake offer name
              </div>
              <div
                className="teaser-sub-title"
              >
                Fake offer type label
              </div>
              <div
                className="mf-infos"
              >
                <span
                  className="mf-status mf-booked"
                >
                  Réservé
                </span>
                <span
                  className="teaser-sub-title"
                >
                  10 km
                </span>
              </div>
            </div>,
            <div
              className="teaser-arrow"
            >
              <Icon
                alt={null}
                className="teaser-arrow-img"
                src={null}
                svg="ico-next-S"
              />
            </div>,
          ],
          "className": "teaser-link",
          "replace": false,
          "to": "https://example.net/decouverte/ME/FA/verso",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": null,
              "className": "teaser-thumb",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="teaser-title"
                >
                  Fake offer name
                </div>,
                <div
                  className="teaser-sub-title"
                >
                  Fake offer type label
                </div>,
                <div
                  className="mf-infos"
                >
                  <span
                    className="mf-status mf-booked"
                  >
                    Réservé
                  </span>
                  <span
                    className="teaser-sub-title"
                  >
                    10 km
                  </span>
                </div>,
              ],
              "className": "teaser-wrapper mf-wrapper",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Fake offer name",
                  "className": "teaser-title",
                },
                "ref": null,
                "rendered": "Fake offer name",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Fake offer type label",
                  "className": "teaser-sub-title",
                },
                "ref": null,
                "rendered": "Fake offer type label",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span
                      className="mf-status mf-booked"
                    >
                      Réservé
                    </span>,
                    <span
                      className="teaser-sub-title"
                    >
                      10 km
                    </span>,
                  ],
                  "className": "mf-infos",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Réservé",
                      "className": "mf-status mf-booked",
                    },
                    "ref": null,
                    "rendered": "Réservé",
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "10 km",
                      "className": "teaser-sub-title",
                    },
                    "ref": null,
                    "rendered": "10 km",
                    "type": "span",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Icon
                alt={null}
                className="teaser-arrow-img"
                src={null}
                svg="ico-next-S"
              />,
              "className": "teaser-arrow",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "alt": null,
                "className": "teaser-arrow-img",
                "src": null,
                "svg": "ico-next-S",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": [Function],
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
